sportsref/nfl/players.py

```diff
@@ -233,6 +242,111 @@ def rushing_and_receiving(self, kind='R'):
         df = sportsref.utils.parse_table(table)
         return df
 
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def defense_and_fumbles(self, kind='R'):
+        """Gets yearly defense/fumble stats for the player.
+
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with defense/fumble stats.
+        """
+        doc = self.get_doc()
+        table = (doc('#all_defense') if kind == 'R'
+                 else doc('#all_defense_playoffs'))
+        df = sportsref.utils.parse_table(table)
+        return df
+
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def kick_and_punt_returns(self, kind='R'):
+        """Gets yearly kick/punt return stats for the player.
+
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with kick/punt return stats.
+        """
+        doc = self.get_doc()
+        table = (doc('#all_returns') if kind == 'R'
+                 else doc('#all_returns_playoffs'))
+        df = sportsref.utils.parse_table(table)
+        return df
+
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def games(self, kind='R'):
+        """Gets yearly games stats for the player.
+
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with games stats.
+        """
+        doc = self.get_doc()
+        table = (doc('#all_games_played') if kind == 'R'
+                 else doc('#all_games_played_playoffs'))
+        df = sportsref.utils.parse_table(table)
+        return df
+
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def kicking_and_punting(self, kind='R'):
+        """Gets yearly kicking/punting stats for the player.
+
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with kicking/punting stats.
+        """
+        doc = self.get_doc()
+        table = (doc('#all_kicking') if kind == 'R'
+                 else doc('#all_kicking_playoffs'))
+        df = sportsref.utils.parse_table(table)
+        return df
+
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def all_scoring(self, kind='R'):
+        """Gets yearly all scoring stats for the player.
+
+        :kind: One of 'R', 'P', or 'B'. Case-insensitive; defaults to 'R'.
+        :returns: Pandas DataFrame with all scoring stats.
+        """
+        doc = self.get_doc()
+        table = (doc('#all_scoring') if kind == 'R'
+                 else doc('#all_scoring_playoffs'))
+        df = sportsref.utils.parse_table(table)
+        return df
+
+
+    @sportsref.decorators.memoize
+    @sportsref.decorators.kind_rpb(include_type=True)
+    def all_annual_stats(self, kind='R'):
```

Comment:
I'll use this function a lot, so I'll thoroughly test it on pretty much every possible player once I get rolling.